Split ZHA "Recommended" from "Other Supported Adapters" in two sections

Hedda opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

Change listing of Silicon Labs and Texas Instruments Zigbee adapters to the top of the Known working Zigbee radio modules so that they are listed before deconz (ConBee/RaspBee) Zigbee adapters, reasons being that both Silabs and TI adapters have many more available options on the market as well as being better supported in ZHA/zigpy than deconz (ConBee/RaspBee) radios.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

Hedda committed:
List Silabs and TI adapters above ConBee/RaspBee

Change listing of Silicon Labs and Texas Instruments Zigbee adapters to the top of the Known working Zigbee radio modules so that they are listed before deCONZ (ConBee/RaspBee) Zigbee adapters, reasons being that both Silabs and TI adapters have many more available options on the market as well as being better supported in ZHA/zigpy than deCONZ (ConBee/RaspBee) Zigbee adapters.

frenck reviewed (changes requested):
Lets just sort the list alphabetically.

Hedda commented:
> Lets just sort the list alphabetically.

That would make it even worse from the suggested compatibility point-of-view since Digi XBee is the least / poorest supported radio by ZHA/zigpy developers today.

It is my belief that many new ZHA end-users will just buy the first adapter that is listed at the very top, so it would be better to have Silicon Labs listed at the top since the upcoming Home Assistant SkyConnect USB Stick (and Home Assistant Yellow) will be based on Silicon Labs radios.

frenck commented:
It is a list of "Known working Zigbee radio modules".
The order should not matter. We should not make it a search game, lets's please sort the list alphabetically.

puddly commented:
The technical reasons for why some some adapters are better-supported than others don't belong in user documentation but this PR is correctly ordering them.  I don't want to encourage people to buy anything but TI, SiLabs, or a Conbee (in that order) and expect a smooth experience with ZHA.

Perhaps we can instead split this into two sections: "Recommended Adapters" and a less-prominent "Other Supported Adapters" (or something similar)?

frenck commented:
Yeah that is fine, but ordering all of these per recommendation will lead to discussions always. Making a first section on recommended adapters and a second one with "others" sounds perfectly fine. It also clearly communicates the recommendation (which currently is implicit)

Hedda commented:
> Perhaps we can instead split this into two sections: "Recommended Adapters" and a less-prominent "Other Supported Adapters" (or something similar)?

That is by the way almost exactly what Zigbee2MQTT official documentation does, as they have compatible adapters sorted in seperate lists for ”Recommended”, ”Not recommended”, and ”Experimental” while under those still grouping the adapters per chip radio type that they are based on, see -> https://www.zigbee2mqtt.io/guide/adapters/

Hedda committed:
Split ZHA supported adapters into two segments

Split ZHA supported adapters into two segments for which are recommended and which are not recommended for new purchase/buyers, which is a little more similar to how Zigbee2MQTT list their supported adapters -> https://www.zigbee2mqtt.io/information/supported_adapters.html

Hedda commented:
OK I have now changed the PR as feedback, splitting lists if ZHA-supported adapters into two segments for which adapters/modules are recommended and which are not recommended for new purchase/buyers, which is a little more similar to how Zigbee2MQTT list their supported adapters -> https://www.zigbee2mqtt.io/information/supported_adapters.html

Hedda commented:
Closing in favor of https://github.com/home-assistant/home-assistant.io/pull/24493

Hedda committed:
Split ZHA Recommended from Other Supported Adapters

Split ZHA Recommended Adapters from Other Supported Adapters into two suggestions as per feedback from @puddly.

> The technical reasons for why some some adapters are better-supported than others don't belong in user documentation but this PR is correctly ordering them. I don't want to encourage people to buy anything but TI, SiLabs, or a Conbee (in that order) and expect a smooth experience with ZHA.
> 
> Perhaps we can instead split this into two sections: "Recommended Adapters" and a less-prominent "Other Supported Adapters" (or something similar)?

With this the "**_Recommended Zigbee radio adapters and modules_**" section with modern/newer hardware from Silicon Labs, Texas Instruments, and Dresden Elektronik, then an "**_Other support but not recommended Zigbee radio adapters or modules_**" section with legacy/older   hardware from Silicon Labs, Texas Instruments, and Dresden Elektronik, as well as Digi XBee and ZiGate.

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -77,25 +77,33 @@ Some other Zigbee coordinator hardware may not support a firmware that is capabl
+  - [ConBee II (a.k.a. ConBee 2) USB adapter from dresden elektronik](https://phoscon.de/conbee2)
+  - [RaspBee II (a.k.a. RaspBee 2) Raspberry Pi Shield from dresden elektronik](https://phoscon.de/raspbee2)
+
+### Other support but not recommended Zigbee radio adapters or modules
```
```suggestion
### Other supported but not recommended Zigbee radio adapters or modules
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -106,7 +114,7 @@ Some other Zigbee coordinator hardware may not support a firmware that is capabl
   - [ZiGate-Ethernet (Ethernet gateway board for PiZiGate)](https://zigate.fr/produit/zigate-ethernet/)
   - [ZiGate + WiFi Pack](https://zigate.fr/produit/zigatev2-pack-wifi/)
-
+  
```
```suggestion

```

frenck committed:
Tweak

frenck reviewed (approved):
Thanks, @Hedda 👍 

../Frenck
